### PR TITLE
Cleanup / Repair

### PR DESCRIPTION
**Pull request by Cathesso**
Fixed bug where Bounds are not being properly loaded after exiting playerBounds. 🔧 
Added Point-Display inside Gameview ℹ️ 
Cleaned up a little. 👏

### CODE REVIEW

File: frontend/src/components/GameController.js [hunk 1]
```diff
@@ -26,7 +26,12 @@ const samNodeIcon = L.icon({
 
 let userMarker = L.marker([27.380583, 33.631839], { icon: userIcon });
 
-export default function GameController({ points, setIsLoading, useMap }) {
+export default function GameController({
```
Comment (jamarob) [inline frontend/src/components/GameController.js:29]: This file really big.. I think you can move some functions to a service file. You have a lot of `let`s that should be `const`. Only use `let` for values that do change!

File: frontend/src/pages/ExplorePage.js [hunk 2]
```diff
@@ -1,10 +1,14 @@
 import styled from "styled-components/macro";
 import StreetMap from "../components/StreetMap";
 
-export default function ExplorePage({ points, setIsLoading }) {
+export default function ExplorePage({ points, setIsLoading, setPoints }) {
```
Comment (jamarob) [inline frontend/src/pages/ExplorePage.js:4]: When passing own all the props you can use the spread operator like so:
```javascript
export default function ExplorePage(props) {
  return (
    <Wrapper>
      <StreetMap {...props} />
    </Wrapper>
  );
```

File: frontend/src/pages/RegisterPage.js [hunk 3]
```diff
@@ -44,16 +44,48 @@ export default function RegisterPage({ setIsLoggedIn }) {
...
+        setPasswordCheckIsNotTheSameAsPassword(false);
+      } else {
+        setPasswordCheckIsNotTheSameAsPassword(true);
+      }
     }
   }, [registerCredentials]); //eslint-disable-line react-hooks/exhaustive-deps
 
+  useEffect(() => {
+    if (usernameAlreadyRegistered !== undefined) {
+      if (usernameAlreadyRegistered === true) {
```
Comment (jamarob) [inline frontend/src/pages/RegisterPage.js:61]:
```suggestion
    if (usernameAlreadyRegistered) {
      if (usernameAlreadyRegistered) {
```

File: frontend/src/pages/RegisterPage.js [hunk 4]
```diff
@@ -44,16 +44,48 @@ export default function RegisterPage({ setIsLoggedIn }) {
...
+        setErrorMessage("Username already taken :C");
+      } else {
+        setErrorMessage("Great username! :)");
+      }
+    }
+  }, [usernameAlreadyRegistered]);
+
+  useEffect(() => {
+    if (passwordIsNotCorrect !== undefined) {
+      if (passwordIsNotCorrect === true) {
```
Comment (jamarob) [inline frontend/src/pages/RegisterPage.js:71]:
```suggestion
    if (passwordIsNotCorrect) {
      if (passwordIsNotCorrect) {
```

File: frontend/src/pages/RegisterPage.js [hunk 5]
```diff
@@ -44,16 +44,48 @@ export default function RegisterPage({ setIsLoggedIn }) {
...
+        setErrorMessage("You need a more secure password. :0");
+      } else {
+        setErrorMessage("This password is perfect! :)");
+      }
+    }
+  }, [passwordIsNotCorrect]);
+
+  useEffect(() => {
+    if (passwordCheckIsNotTheSameAsPassword !== undefined) {
+      if (passwordCheckIsNotTheSameAsPassword === true) {
```
Comment (jamarob) [inline frontend/src/pages/RegisterPage.js:81]: see above ;)

**Review by jamarob — COMMENTED** [review]
Good refactoring! Left a few comments.

File: frontend/src/pages/ExplorePage.js (hunk 2 above)
Comment (Cathesso) [reply frontend/src/pages/ExplorePage.js:4]: That's nifty, thanks!

File: frontend/src/pages/RegisterPage.js (hunk 3 above)
Comment (Cathesso) [reply frontend/src/pages/RegisterPage.js:61]: It works, if I change only the second variable. The first one needs to stay as "!== undefined" for my function to work properly :)

File: frontend/src/pages/RegisterPage.js (hunk 4 above)
Comment (Cathesso) [reply frontend/src/pages/RegisterPage.js:71]: Same as above.

File: frontend/src/components/GameController.js (hunk 1 above)
Comment (Cathesso) [reply frontend/src/components/GameController.js:29]: I put that on my bucketlist! Thanks for the input! :)

**Pull request merged**